### PR TITLE
Updated Hazelcast Platform Operator information [CN-240]

### PR DESCRIPTION
**Pull request by alparslanavci**
- Changed the naming of Platform Operator
- Removed the old getting started section and provided a link to the new operator docs.

### CODE REVIEW

File: docs/modules/deploy/pages/deploying-in-kubernetes.adoc [hunk 1]
```diff
@@ -36,12 +36,17 @@ https://helm.sh/[Helm^] is a package manager for Kubernetes. Hazelcast is distri
 See the https://github.com/hazelcast/charts[Hazelcast Helm Charts documentation^]
 for more details.
 
+== Hazelcast Platform Operator
+
+Kubernetes Operators are software extensions to Kubernetes which help you create and manage your applications. Hazelcast Platform Operator automates common management tasks such as configuring, creating, scaling, and recovering Hazelcast clusters on Kubernetes and Red Hat OpenShift. By taking care of manual deployment and life-cycle management, Hazelcast Platform Operator makes it simpler to work with Hazelcast clusters.
```
Comment (hasancelik) [inline docs/modules/deploy/pages/deploying-in-kubernetes.adoc:41]: `tasks such as configuring, creating, scaling, and recovering`

Currently, we do not have proper recovery functionality so I think we should not mention it here. WDYT?

Comment (alparslanavci) [reply docs/modules/deploy/pages/deploying-in-kubernetes.adoc:41]: Agreed 👍

**Review by hasancelik — APPROVED** [review]

File: docs/modules/deploy/pages/deploying-in-kubernetes.adoc [hunk 2]
```diff
@@ -36,12 +36,17 @@ https://helm.sh/[Helm^] is a package manager for Kubernetes. Hazelcast is distri
 See the https://github.com/hazelcast/charts[Hazelcast Helm Charts documentation^]
 for more details.
 
+== Hazelcast Platform Operator
+
+Kubernetes Operators are software extensions to Kubernetes which help you create and manage your applications. Hazelcast Platform Operator automates common management tasks such as configuring, creating, and scaling Hazelcast clusters on Kubernetes and Red Hat OpenShift. By taking care of manual deployment and life-cycle management, Hazelcast Platform Operator makes it simpler to work with Hazelcast clusters.
+
+See the https://docs.hazelcast.com/hazelcast-operator/latest/[Hazelcast Platform Operator Documentation] for more information and find details about deploying a Hazelcast cluster in Kubernetes and also connecting clients outside Kubernetes.
```
Comment (alparslanavci) [inline docs/modules/deploy/pages/deploying-in-kubernetes.adoc:43]: The link here is not correct, it should be `https://docs.hazelcast.com/operator/latest/`

File: docs/modules/deploy/partials/nav.adoc [hunk 3]
```diff
@@ -16,6 +16,6 @@
 *** xref:deploy:deploying-in-kubernetes.adoc[Overview]
 *** xref:deploy:configuring-kubernetes.adoc[]
 *** xref:deploy:scaling-and-upgrading-kubernetes.adoc[]
-*** xref:deploy:deploying-with-operator.adoc[]
+*** xref:operator:root:index.adoc[Hazelcast Platform Operator]
```
Comment (alparslanavci) [inline docs/modules/deploy/partials/nav.adoc:19]: I didn't get the reason for this change? "Hazelcast Platform Operator" in this section doesn't point to anywhere.

**Pull request merged**